src/common/com/intellij/plugins/haxe/model/HaxeEnumModel.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright 2018 Ilya Malanin
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intellij.plugins.haxe.model;
+
+import com.intellij.plugins.haxe.lang.psi.*;
+import com.intellij.util.SmartList;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class HaxeEnumModel extends HaxeClassModel implements IHaxeEnumModel {
+  public HaxeEnumModel(@NotNull HaxeEnumDeclaration haxeClass) {
+    super(haxeClass);
+  }
+
+  public HaxeEnumDeclaration getEnumDeclaration() {
+    return (HaxeEnumDeclaration)haxeClass;
+  }
+
+  @Override
+  public HaxeClassModel getParentClass() {
+    return null;
+  }
+
+  @Override
+  public List<HaxeClassReferenceModel> getInterfaceExtendingInterfaces() {
+    return Collections.emptyList();
+  }
+
+  @Override
+  public List<HaxeClassReferenceModel> getImplementingInterfaces() {
+    return Collections.emptyList();
+  }
+
+  @Override
+  public boolean isClass() {
+    return false;
+  }
+
+  @Override
+  public boolean isInterface() {
+    return false;
+  }
+
+  @Override
+  public boolean isEnum() {
+    return true;
+  }
+
+  @Override
+  public boolean isTypedef() {
+    return false;
+  }
+
+  @Nullable
+  @Override
+  public HaxeTypeOrAnonymous getUnderlyingType() {
+    return null;
+  }
+
+  @Override
+  public List<HaxeType> getAbstractToList() {
+    return Collections.emptyList();
+  }
+
+  @Override
+  public List<HaxeType> getAbstractFromList() {
+    return Collections.emptyList();
+  }
+
+  @Override
+  public boolean hasMethod(String name) {
+    return false;
+  }
+
+  @Override
+  public boolean hasMethodSelf(String name) {
+    return false;
+  }
+
+  @Override
+  public HaxeMethodModel getMethodSelf(String name) {
+    return null;
+  }
+
+  @Override
+  public HaxeMethodModel getConstructorSelf() {
+    return null;
+  }
+
+  @Override
+  public HaxeMethodModel getConstructor() {
+    return null;
+  }
+
+  @Override
+  public boolean hasConstructor() {
+    return false;
+  }
+
+  @Override
+  public HaxeMethodModel getParentConstructor() {
+    return null;
+  }
+
+  @Override
+  public HaxeMemberModel getMember(@NotNull final String name) {
+    return getValue(name);
+  }
+
+  @Override
+  public List<HaxeMemberModel> getMembers() {
+    return getValuesStream().collect(Collectors.toList());
```

Comment:
Ehm? `getValuesStream` is a source of nodes, like it was before. I've separated creating of stream and collecting items from it into different methods. Because sometimes it's much easier and profitable to work with stream that have a lazy evaluation, and premade items list often redundant.

Reply:
I missed the .map() call in getValuesStream().  I read the code and saw, basically, Stream.fromList(valuesList).toList().  My mistake here.